app/uk/gov/hmrc/cipphonenumberverification/services/NotificationService.scala

```diff
@@ -55,20 +57,25 @@ class NotificationService @Inject()(govNotifyUtils: GovNotifyUtils, auditService
         case "temporary-failure" => (107, "Message was unable to be delivered by the network provider")
         case "technical-failure" => (108, "There is a problem with the notification vendor")
       }
-      auditService.sendExplicitAuditEvent(AuditType.PHONE_NUMBER_VERIFICATION_DELIVERY_RESULT_REQUEST.toString,
+      auditService.sendExplicitAuditEvent(PhoneNumberVerificationDeliveryResultRequest,
         VerificationDeliveryResultRequestAuditEvent(phoneNumber, passcode, notificationId, deliveryStatus))
       Ok(Json.toJson(NotificationStatus(code, message)))
     }
 
     def failure(err: UpstreamErrorResponse) = {
       err.statusCode match {
         case NOT_FOUND =>
-          logger.warn("Notification ID not found")
-          auditService.sendExplicitAuditEvent(AuditType.PHONE_NUMBER_VERIFICATION_DELIVERY_RESULT_REQUEST.toString,
+          logger.warn("Notification Id not found")
+          auditService.sendExplicitAuditEvent(PhoneNumberVerificationDeliveryResultRequest,
             VerificationDeliveryResultRequestAuditEvent(NO_DATA_FOUND, NO_DATA_FOUND, notificationId, NO_DATA_FOUND))
           NotFound(Json.toJson(ErrorResponse(Codes.NOT_FOUND, NO_DATA_FOUND)))
+        case BAD_REQUEST =>
+          logger.warn("Notification Id not valid")
+          BadRequest(Json.toJson(ErrorResponse(VALIDATION_ERROR, "Enter a valid notification Id")))
+        case FORBIDDEN =>
+          logger.warn(err.message)
+          ServiceUnavailable(Json.toJson(ErrorResponse(EXTERNAL_API_FAIL, "External server currently unavailable")))
         case _ =>
-          //TODO: Do we need a separate ticket to handle other errors?
```

Comment:
Regarding the _ case, should that be a 500 instead of a HttpEntity.NoEntity?

Reply:
as discussed leave as is

Reply:
As I mentioned we will still send the error.status code